Sources/ArcGISToolkit/Components/FeatureTemplatePicker/FeatureTemplatePicker.swift

```diff
@@ -0,0 +1,334 @@
+    ///   - includeNonCreatableFeatureTemplates: Include feature templates from tables where features cannot be created.
+    public init(geoModel: GeoModel, selection: Binding<FeatureTemplateInfo?>, includeNonCreatableFeatureTemplates: Bool = false) {
+        _model = StateObject(
+            wrappedValue: Model(
+                geoModel: geoModel,
+                includeNonCreatableFeatureTemplates: includeNonCreatableFeatureTemplates
+            )
+        )
+        _selection = selection
+    }
+    
+    public var body: some View {
+        Group {
+            if model.isGeneratingFeatureTemplates {
+                ProgressView()
+            } else if model.showContentUnavailable {
+                if #available(iOS 17.0, *) {
+                    ContentUnavailableView(
+                        String.noFeatureTemplatesTitle,
+                        systemImage: "mappin.slash.circle",
+                        description: Text(String.noFeatureTemplatesDetail)
+                    )
+                } else {
+                    // Fallback on earlier versions
+                    Text(String.noFeatureTemplatesDetail)
+                }
+            } else {
+                List {
+                    ForEach(model.featureTemplateSections) { section in
+                        Section(section.table.displayName) {
+                            ForEach(section.infos) { info in
+                                FeatureTemplateView(info: info, selection: _selection)
+                            }
+                        }
+                    }
+                }
+                .searchable(text: $model.searchText, prompt: String.searchTemplatesPrompt)
+                .overlay {
+                    if model.showNoTemplatesFound {
+                        if #available(iOS 17.0, *) {
+                            ContentUnavailableView(
+                                String.noFeatureTemplatesFoundTitle,
+                                systemImage: "magnifyingglass.circle",
+                                description: Text(String.noFeatureTemplatesFoundDetail)
+                            )
+                        } else {
+                            // Fallback on earlier versions
+                            Text(String.noFeatureTemplatesFoundDetail)
+                        }
+                    }
+                }
+            }
+        }
+        .task {
+            await model.generateFeatureTemplates()
+        }
+    }
+}
+
+private extension String {
+    static var noFeatureTemplatesTitle: String {
+        String(
+            localized: "No Feature Templates",
+            bundle: .toolkitModule,
+            comment: """
+                 A title for showing a view that tells the user there are no feature templates.
+                 """
+        )
+    }
+    static var noFeatureTemplatesDetail: String {
+        String(
+            localized: "There are no feature templates available for this map.",
```

Comment:
Should we generalize this for use with scenes?

```suggestion
            localized: "There are no feature templates available.",
```